lib/assets.js

```diff
@@ -0,0 +1,244 @@
+  }
+  this.globOpts.cwd = this.srcPath; // cannot be overridden by globOpts
+  this.globOpts.root = this.srcPath; // cannot be overridden by globOpts
+}
+
+function quoted(string) {
+  return string ? '"' + string + '"' : string;
+}
+
+function resolveModuleName(assetPathEntry, defaultName) {
+  return assetPathEntry.name || defaultName;
+}
+
+AssetPathEntry.prototype = {
+  defaultGlobOpts: {
+    follow: true,
+    nodir: true
+  },
+  asRegistrationString: function(defaultName) {
+    var files = glob.sync(this.pattern, this.globOpts);
+    var assetMap = [];
+    var self = this;
+    files.forEach(function(file) {
+      // TODO: handle windows error
+      var uri = self.httpPrefix;
+      var moduleName = resolveModuleName(self, defaultName);
+
+      // TODO: do this in a helper function
+      if (uri && moduleName) {
+        uri = uri + "/" + moduleName;
+      } else if (moduleName) {
+        uri = moduleName;
+      }
+
+      if (uri) {
+        uri = uri + "/" + file;
+      } else {
+        uri = file;
+      }
+
+      assetMap.push('"' + file + '": (filepath: "' + path.join(self.srcPath, file) + '", uri: "' + uri + '")');
+    });
+    return "@include register-assets(" +
+             (quoted(resolveModuleName(this, defaultName)) || "null") + ", " +
+             "(" + assetMap.join(",\n  ") + "));";
+  },
+  toString: function() {
+    return "<" + this.pattern + " within " + this.sourcePath + ">";
+  }
+};
+
+
+function AssetCollection() {
+  this.assetPath = [];
+}
+
+AssetCollection.prototype = {
+  /* See: AssetPathEntry */
+  addSource: function(srcPath, opts) {
+    this.assetPath.push(new AssetPathEntry(srcPath, opts));
+    return this;
+  },
+  asAssetImport: function (defaultName) {
+    var scssString = ['@import "eyeglass/assets";'];
+    this.assetPath.forEach(function(pathEntry) {
+      scssString.push(pathEntry.asRegistrationString(defaultName));
+    });
+    var importString = scssString.join("\n");
+    return importString;
+  }
+};
+
+function chainableExport(srcPath, opts) {
+  return this.addSource(srcPath, opts);
+}
+
+function newExportableAssetCollection(srcPath, opts) {
+    var assets = new AssetCollection();
+    assets.export = chainableExport;
+    return assets.export(srcPath, opts);
+}
+
+function resolveAsset(sass, sassUtils, registeredAssetsMap, relativePathString) {
+  registeredAssetsMap = sassUtils.handleEmptyMap(registeredAssetsMap);
+  sassUtils.assertType(registeredAssetsMap, "map");
+  sassUtils.assertType(relativePathString, "string");
+
+  var relativePath = unquote(relativePathString.getValue());
+  var registeredAssets = sassUtils.castToJs(registeredAssetsMap);
+
+  var appAssets = registeredAssets.coerce.get(null);
+  if (appAssets) {
+    // XXX sassUtils.assertType(appAssets, "map");
+    var appAsset = appAssets.coerce.get(relativePath);
+    if (appAsset) {
+      return appAsset;
+    }
+  }
+
+  var segments = relativePath.split("/");
```

Comment:
```js
var segments = relativePath.split(path.sep):
```

Should work, but you might want to take another look at this section of figuring out the module name.

----

Perhaps split it out into a separate module.